Source/EventAnalysis/KTOverlappingTrackClustering.cc

```diff
@@ -0,0 +1,406 @@
+            newTrack.SetSlope( (newTrack.GetEndFrequency() - newTrack.GetStartFrequency())/(newTrack.GetEndTimeInRunC() - newTrack.GetStartTimeInRunC()));
+
+        }
+        if (oldTrack.GetEndTimeInRunC() > newTrack.GetEndTimeInRunC())
+        {
+            newTrack.SetEndTimeInRunC( oldTrack.GetEndTimeInRunC());
+            newTrack.SetEndFrequency( oldTrack.GetEndFrequency());
+            newTrack.SetEndTimeInRunCSigma( oldTrack.GetEndTimeInRunCSigma());
+            newTrack.SetEndFrequencySigma( oldTrack.GetEndFrequencySigma());
+            newTrack.SetSlope( (newTrack.GetEndFrequency() - newTrack.GetStartFrequency())/(newTrack.GetEndTimeInRunC() - newTrack.GetStartTimeInRunC()));
+
+        }
+        //newTrack.SetSlope( (newTrack.GetEndFrequency() - newTrack.GetStartFrequency())/(newTrack.GetEndTimeInRunC() - newTrack.GetStartTimeInRunC()));
+        newTrack.SetTotalPower( newTrack.GetTotalPower() + oldTrack.GetTotalPower());
+        newTrack.SetTimeLength( newTrack.GetEndTimeInRunC() - newTrack.GetStartTimeInRunC());
+
+        newTrack.SetSlopeSigma( sqrt(newTrack.GetSlopeSigma()*newTrack.GetSlopeSigma() + oldTrack.GetSlopeSigma()*oldTrack.GetSlopeSigma()));
+    }
+
+/*
+    bool KTOverlappingTrackClustering::DoTheyMatch(KTProcessedTrackData& Track1, KTProcessedTrackData& Track2)
+    {
+        bool slope_condition1 = std::abs(Track1.GetEndFrequency()+Track1.GetSlope()*(Track2.GetStartTimeInRunC()-Track1.GetEndTimeInRunC()) - Track2.GetStartFrequency())<fMaxTrackWidth;
+        bool slope_condition2 = std::abs(Track2.GetStartFrequency()-Track2.GetSlope()*(Track2.GetStartTimeInRunC()-Track1.GetEndTimeInRunC()) - Track1.GetEndFrequency())<fMaxTrackWidth;
+        bool time_gap_in_line = Track1.GetEndTimeInRunC() < Track2.GetStartTimeInRunC();
+        bool gap_smaller_than_limit = std::abs(Track2.GetStartTimeInRunC() - Track1.GetEndTimeInRunC())<fTimeGapTolerance;
+
+        if (time_gap_in_line and gap_smaller_than_limit and (slope_condition1 or slope_condition2))
+        {
+            return true;
+        }
+
+        slope_condition1 = std::abs(Track2.GetEndFrequency()+Track2.GetSlope()*(Track1.GetStartTimeInRunC()-Track2.GetEndTimeInRunC()) - Track1.GetStartFrequency())<fMaxTrackWidth;
+        slope_condition2 = std::abs(Track1.GetStartFrequency()-Track1.GetSlope()*(Track1.GetStartTimeInRunC()-Track2.GetEndTimeInRunC()) - Track2.GetEndFrequency())<fMaxTrackWidth;
+        time_gap_in_line = Track2.GetEndTimeInRunC() < Track1.GetStartTimeInRunC();
+        gap_smaller_than_limit = std::abs(Track1.GetStartTimeInRunC() - Track2.GetEndTimeInRunC())<fTimeGapTolerance;
+
+        if (time_gap_in_line and gap_smaller_than_limit and (slope_condition1 or slope_condition2))
+        {
+            return true;
+        }
+
+    return false;
+    }
+*/
+    bool KTOverlappingTrackClustering::DoTheyOverlap(KTProcessedTrackData& Track1, KTProcessedTrackData& Track2)
```

Comment:
1. Variable names that need fixing: `Track1`, `Track2`

1. If there are calculations in common between the different conditions that you calculate, it's often more efficient to do them in advance than do them multiple times.

1. I think this method can be const